### ISSUE TITLE
Smart default mapping for controller

### ISSUE THREAD

**Issue by bilhox** [pygame-community/pygame-ce#3088]
Hello,

While playing with ``_sdl2.controller``, I found that for all controllers I had, default mapping was wrong. Okay, we propose a way for users to set their mapping, but wouldn't be ideal for the comfort of users to propose smart default mapping that matches the button/axis constants proposed by the ``controller`` module ?

What I can propose is to do like some people in the SDL community do, keeping a database of the most used controllers in the world with their default mapping.

What do you think ?

**Comment by damusss** [comment]
why not, yeah. maybe enabled by some flag

**Comment by Starbuck5** [comment]
Wdym it was wrong? Isn’t the whole point of SDLs controller system to map everything to be like an Xbox controller?

**Comment by bilhox** [comment]
> Wdym it was wrong? Isn’t the whole point of SDLs controller system to map everything to be like an Xbox controller?

Yes! Sorry for the misunderstanding also, I thought it would also take in account controllers like SNES Controller which looks like this : 
![image](https://github.com/user-attachments/assets/edd2cc16-8345-4583-b69d-a50a4d89ae15)

It just doesn't have sticks and shoulders button/trigger.

The button A and B are inverted on my nintendo joycons as well. (when they are combined*)

**Comment by bilhox** [comment]
Someone shared me this on the SDL discord, it might be worth taking a look at it : 
https://github.com/mdqinc/SDL_GameControllerDB/tree/master

**Comment by zoldalma999** [comment]
> While playing with _sdl2.controller, I found that for all controllers I had, default mapping was wrong. Okay, we propose a way for users to set their mapping, but wouldn't be ideal for the comfort of users to propose smart default mapping that matches the button/axis constants proposed by the controller module ?

That is the point of the SDL_Controller (or SDL_Gamepad in sdl3) module. We should not need to keep our own database of controller layouts. It should just work. And if it does not, we make a PR to sdl.

With that said, we could vendor in the one you linked (https://github.com/mdqinc/SDL_GameControllerDB/tree/master). Or mention it in the docs.

**Comment by bilhox** [comment]
Looks like SDL has a function for loading mappings from a file, could be interesting to implement it. (Which would close this issue)

**Comment by JiffyRob** [comment]
I actually requested that the mapping adding function be bound in the contributing discord.  My use case is that I have a USB gamepad that I found buried in a closet that looks like a PS2 controller but doesn't register as an SDL2 controller.  I'd like to be able to test using it.
